docs/source/en/pipeline_tutorial.mdx

```diff
@@ -33,101 +33,295 @@ While each task has an associated [`pipeline`], it is simpler to use the general
+For instance this pipeline also handles `chunk_length_s` which is helpful to work on 
+really long audio files which the models cannot handle on their own. This is helpful when
+subtitling entire movies or hour-long videos.
+
+
+Cannot you find a parameter which would really help you out ?
+[Request it](https://github.com/huggingface/transformers/issues/new?assignees=&labels=feature&template=feature-request.yml)
+
+
+## Using pipelines on a dataset
+
+So you have a large dataset and want to run inference on it to get something like a
+metric for it, perfect !
+
+The easiest and more generic way we recomment is using an iterator.
 
 ```py
->>> generator(
-...     "Three Rings for the Elven-kings under the sky, Seven for the Dwarf-lords in their halls of stone"
-... )  # doctest: +SKIP
-[{'generated_text': 'Three Rings for the Elven-kings under the sky, Seven for the Dwarf-lords in their halls of stone, Seven for the Dragon-lords (for them to rule in a world ruled by their rulers, and all who live within the realm'}]
+def data():
+    for i in range(1000):
+        yield f"My example {i}"
+
+
+pipe = pipe(model="gpt2", device=0)
+generated_characters = 0
+for out in pipe(data()):
+    generated_characters += len(out["generated_text"])
 ```
 
-## Audio pipeline
+The iterator `data()` will yield each result. The pipeline will automatically
+recognize that the input is iterable, and will start fetching the data while
+it continues to process it on the GPU. This uses [DataLoader](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader) under the hood.
+This is important so you don't have to allocate memory for the whole dataset
+and you can feed the GPU as fast as possible.
 
-The [`pipeline`] also supports audio tasks like audio classification and automatic speech recognition.
+Since batching could speed things up, using `batch_size` here and tuning it could be important.
 
-For example, let's classify the emotion in this audio clip:
+Turns out we also have a fantasic library for [datasets](https://github.com/huggingface/datasets/).
+which makes this even simpler.
 
 ```py
->>> from datasets import load_dataset
->>> import torch
+# KeyDataste is a util that will just output the item we're interested in.
+from transformers.pipelines.pt_utils import KeyDataset
+
+pipe = pipeline(model="hf-internal-testing/tiny-random-wav2vec2", device=0)
+dataset = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation[:10]")
 
->>> torch.manual_seed(42)  # doctest: +IGNORE_RESULT
->>> ds = load_dataset("hf-internal-testing/librispeech_asr_demo", "clean", split="validation")
->>> audio_file = ds[0]["audio"]["path"]
+for out in pipe(KeyDataset(dataset["audio"])):
+   print(out)
 ```
 
-Find an [audio classification](https://huggingface.co/models?pipeline_tag=audio-classification) model on the Model Hub for emotion recognition and load it in the [`pipeline`]:
+
+## Using pipelines for a webserver
```

Comment:
This is too advanced for this tutorial. It can go in another page however as this is certainly of interest for some users!